### ISSUE TITLE
Change hints schema.

### ISSUE THREAD

**Issue by danielballan** [bluesky/databroker#462]
Per discussion with @ronpandolfi, @ihumphrey, and @DylanMcReynolds, it would help to make the hints more self-describing if we changed entries like

```json
["primary", "fccd_image"]
```

to

```
{"stream_name": "primary", "field": "fccd_image"}
```

**Comment by tacaswell** [comment]
We don't expect to pick up any more fields in this value (as that would imply we have folded on the "event/event pages are lowest level thing that can have structure" line) so we can define in the schema what the values mean without any ambiquity.

On the other hand we want this to be consistent with how we specify data keys in hints and in the "techniques" / "experiments" entry in the start document which I could imagine we might want to be able to specify a slice so that would be a third or more entries into this structure.

We should also consider going the other direction and encoding this as a uri ex `primary.fccd_image` or `primary/fccd_image`.

**Comment by danielballan** [comment]
Yes, I think it's unlikely to grow. The motivation was to make it more _self-describing_.
